src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -277,4 +199,114 @@ private function getConstantDeclaringClass(string $constantName, ReflectionClass
 
         return $parentClass ? $this->getConstantDeclaringClass($constantName, $parentClass) : null;
     }
+
+    /**
+     * @return callable[] indexed by node class name
+     */
+    private static function loadEvaluators() : array
```

Comment:
I should probably split this out to an external class...

Reply:
Thinking twice: no, that's probably not such a good idea. Having it split out for the sake of it would probably just increase complexity right now, and cause a cyclic dependency.